internal/mode/static/config_updater.go

```diff
@@ -100,6 +58,36 @@ func updateControlPlane(
 		)
 	}
 
-	// set the log level
-	return logLevelSetter.SetLevel(string(*controlConfig.Logging.Level))
+	level := *controlConfig.Logging.Level
+
+	if err := validateLogLevel(level); err != nil {
+		return err
+	}
+
+	if err := logLevelSetter.SetLevel(string(level)); err != nil {
+		return field.Invalid(
+			field.NewPath("logging.level"),
+			level,
+			err.Error(),
+		)
+	}
+
+	return nil
+}
+
+func validateLogLevel(level ngfAPI.ControllerLogLevel) error {
```

Comment:
CRD is doing this validation for us. Per our other offline discussions, do we need to do it again?

Reply:
This change was motivated by @pleshakov's comment here: https://github.com/nginxinc/nginx-gateway-fabric/pull/1347#discussion_r1439938880


Initially, I was thinking I could just change the `logLevelSetter` to return the errors from `SetLevel` and wrap that error in the `field.NotSupported` error, but `field.NotSupported` doesn't accept an error or message, it accepts `(field *Path, value interface{}, validValues []T)`. I didn't want to ignore the error returned from `SetLevel`, so I decided to separate out the validation from the `SetLevel` call. 

I don't think we need to be re-applying the CRD validation, but this seemed like a unique case.

Alternatively, I could remove this function and return `field.Invalid` instead of `field.NotSupported` when `SetLevel` fails.

Reply:
Makes sense. field.NotSupported is the better error in this case than field.Invalid.